### ISSUE TITLE
Does not protect against DST changes / clock going backwards

### ISSUE THREAD

**Issue by RobThree** [rvhuang/snowflake-lite#1]
When clock goes back ([DST](https://en.wikipedia.org/wiki/Daylight_saving_time) change) there's a real possibility of generating duplicate ID's. Suggestion: use a `strictly increasing (monotonic) time source` ([this one](https://msdn.microsoft.com/en-us/library/windows/desktop/dn553408%28v=vs.85%29.aspx) for example).

Full disclosure: I'm the developer of [IdGen](https://github.com/RobThree/IdGen) which is a similar project.

**Comment by rvhuang** [comment]
@RobThree does DST affect `DateTime.UtcNow`?

**Comment by RobThree** [comment]
No, but any corrections made by [NTP](https://en.wikipedia.org/wiki/Network_Time_Protocol) for example (which Windows does by default) _will_ affect the time (possibly several times a day) and then there's [leap seconds](https://en.wikipedia.org/wiki/Leap_second) and all kinds of other horrible messy things people "invent" to keep track of "time". Wallclock time, which `DateTime.UtcNow` (and `DateTime.Now`) is, is not usable for this kind of applications.
